### ISSUE TITLE
SFTPName._format doesn't handle string args correctly

### ISSUE THREAD

**Issue by zentarim** [ronf/asyncssh#407]
Hello all! Got this traceback trying to interate through remote directory having `type(path) == str`:

```
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 371, in getMessage
    msg = str(self.msg)
  File "/home/user/test_asyncssh/venv/lib/python3.8/site-packages/asyncssh/misc.py", line 304, in __str__
    return ', '.join('%s: %s' % (k, v) for k, v in values if v is not None)
  File "/home/user/test_asyncssh/venv/lib/python3.8/site-packages/asyncssh/misc.py", line 304, in <genexpr>
    return ', '.join('%s: %s' % (k, v) for k, v in values if v is not None)
  File "/home/user/test_asyncssh/venv/lib/python3.8/site-packages/asyncssh/misc.py", line 301, in <genexpr>
    values = ((k, self._format(k, getattr(self, k)))
  File "/home/user/test_asyncssh/venv/lib/python3.8/site-packages/asyncssh/sftp.py", line 996, in _format
    return v.decode('utf-8', errors='replace') or None
AttributeError: 'str' object has no attribute 'decode'

```

Test script:
```
import logging
from asyncio import run

from asyncssh import connect, SFTPClient, SSHClientConnection


async def main(host: str, port: int, user: str, password: str):
    connection_cm = connect(host, port, username=user, password=password, known_hosts=None)
    async with connection_cm as ssh:  # type: SSHClientConnection
        async with ssh.start_sftp_client() as sftp:  # type: SFTPClient
            async for file in sftp.scandir(path='.'):
                logging.info(file)


if __name__ == '__main__':
    logging.basicConfig(level=logging.DEBUG)
    run(main("127.0.0.1", 22, "xan", "1111111111"))
```

This crude patch makes the job done:
```
--- sftp.py.bak 2021-09-11 07:33:56.248813890 +0300
+++ sftp.py     2021-09-11 07:34:12.208934413 +0300
@@ -992,6 +992,8 @@
 
         if k == 'attrs':
             return str(v) or None
+        elif isinstance(v, str):
+            return v
         else:
             return v.decode('utf-8', errors='replace') or None

```

AsyncSSH version: 2.7.1

Cheers!

**Comment by ronf** [comment]
Thanks for the report! I decided to do a bit of additional simplification of the code in question, but it should fix the issue you reported. It is now checked into the "develop" branch as commit a8b6c51, and this will be included in the next release.

**Comment by zentarim** [comment]
> Thanks for the report! I decided to do a bit of additional simplification of the code in question, but it should fix the issue you reported. It is now checked into the "develop" branch as commit [a8b6c51](https://github.com/ronf/asyncssh/commit/a8b6c5116f209377e876eee247445030e4222c6a), and this will be included in the next release.

Appreciated! Thanks a lot for the fast answer!

**Comment by ronf** [comment]
This fix is now released in AsyncSSH 2.7.2. Please let me know if you are seeing any issues with the new version!

**Comment by zentarim** [comment]
Yep, it works. Thanks a lot!
